### ISSUE TITLE
Select field loses focus after clearing selection

### ISSUE THREAD

**Issue by blackbaud-conorwright** [blackbaud/skyux-select-field#24]
### Expected behavior
Select field should retain focus

### Actual behavior
The focused clear button is erased from the document and focus defaults to the body.
This can be disorienting for accessible users to have their position in the document shifted to root suddenly.

This also causes the blur event to fire twice in an instant. Once for the value change and again for the component losing focus. (this is what breaks our stuff and we have some fixing to do on our end too)

### Steps to reproduce
1. In singlepicker mode, Select a value
2. Now clear the value using the "x" button
3. See that the "x" button is no longer rendered
4. Open the console and type `document.activeElement` to see that the focused element is the body and not the sky-select-field (or any of its children)

### Severity
Medium

### Impact
Low

**Comment by Blackbaud-TrevorBurch** [comment]
In `3.1.1`
